MantidQt/CustomInterfaces/src/Indirect/ConvFit.cpp

```diff
@@ -222,6 +209,72 @@ void ConvFit::setup() {
   updatePlotOptions();
 }
 
+/** Setup FABADA minimizer options
+*
+*/
+void ConvFit::initFABADAOptions() {
```

Comment:
I've checked and I can rewrite this method as:

```
void ConvFit::initFABADAOptions() {

  // Map from minimizer property name to interface property name
  std::map<std::string, std::string> interfaceNames = {
      {"FABADAOutputChain", "Output Chain"},
      {"FABADAChainLength", "Chain Length"},
      {"FABADAConvergenceCriteria", "Convergence Criteria"},
      {"FABADAJumpAcceptanceRate", "Acceptance Rate"},
      {"FABADAStepsBetweenValues", "Steps Between Values"},
      {"FABADAInactiveConvergenceCriterion", "Inactive Convergence Criterion"},
      {"FABADASimAnnealingApplied", "Sim Annealing Applied"},
      {"FABADAMaximumTemperature", "Maximum Temperature"},
      {"FABADANumRefrigerationSteps", "Num Refrigeration Steps"},
      {"FABADASimAnnealingIterations", "Sim Annealing Iterations"},
      {"FABADAOverexploration", "Overexploration"},
      {"FABADANumberBinsPDF", "Number Bins PDF"}};

  m_properties["FABADA"] = m_grpManager->addProperty("Bayesian");
  m_properties["UseFABADA"] = m_blnManager->addProperty("Use FABADA");
  m_properties["FABADA"]->addSubProperty(m_properties["UseFABADA"]);

  const auto &minimizer =
      Mantid::API::FuncMinimizerFactory::Instance().createMinimizer("FABADA");

  for (const auto &prop : minimizer->getProperties()) {

    if (prop->type() == "boolean") {

      const auto name = QString::fromStdString("FABADA" + prop->name());
      m_properties[name] = m_blnManager->addProperty(
          QString::fromStdString(interfaceNames[prop->name()]));
      m_blnManager->setValue(m_properties[name],
                             (prop->getDefault() == "1") ? true : false);

    } else if (prop->type() == "number") {

      const auto name = QString::fromStdString("FABADA" + prop->name());
      m_properties[name] = m_dblManager->addProperty(
          QString::fromStdString(interfaceNames[prop->name()]));
      m_dblManager->setValue(m_properties[name],
                             boost::lexical_cast<double>(prop->getDefault()));

    } else if (prop->type() == "MatrixWorkspace" ||
               prop->type() == "TableWorkspace") {
      // Don't expose these properties
    } else {
      // Assume size_t

      const auto name = QString::fromStdString("FABADA" + prop->name());
      m_properties[name] = m_dblManager->addProperty(
          QString::fromStdString(interfaceNames[prop->name()]));
      m_dblManager->setValue(m_properties[name],
                             boost::lexical_cast<double>(prop->getDefault()));
      m_dblManager->setDecimals(m_properties[name], 0);
    }
  }

  // Advanced options
  m_properties["FABADAAdvanced"] = m_blnManager->addProperty("Advanced");
  m_blnManager->setValue(m_properties["FABADAAdvanced"], false);

  m_cfTree->addProperty(m_properties["FABADA"]);
}
```

The reason why I need the `interfaceNames` map is because some properties are not called the same in the interface and the minimizer (`JumpAcceptanceRate` is for instance `Acceptance Rate` in the interface, and the others have white spaces). 
On the other hand, I need to deal with property types, as some of them are added to `m_blnManger` and others are added to `m_dblManager`, whereas `MatrixWorkspace` and `TableWorkspace` should not be exposed. If new `double`, `boolean` or `integer` properties are added to the minimizer, they will automatially get added to the interface, but if someone adds other types of properties, for instance a vector of numbers, they'll have to add another `if` statement.

Do you think I should change the method? I find the code more complicated, although perhaps there's a simpler approach.